sw/device/lib/dif/dif_hmac.c

```diff
@@ -0,0 +1,339 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_hmac.h"
+
+#include "hmac_regs.h"  // Generated.
+#include "sw/device/lib/base/memory.h"
+
+#define HMAC_HASH_SIZE_BITS 256u
+#define HMAC_KEY_REG_COUNT (HMAC_HASH_SIZE_BITS / 32)
+#define HMAC_DIGEST_REG_COUNT HMAC_KEY_REG_COUNT
+
+/**
+ * Load |key| into |hmac|.
+ */
+static void load_key(const dif_hmac_t *hmac, const uint8_t *key) {
+  for (int i = 0; i < HMAC_KEY_REG_COUNT; ++i) {
+    ptrdiff_t offset = i * sizeof(uint32_t);
+    uint32_t key_part;
+
+    memcpy(&key_part, key + offset, sizeof(key_part));
+
+    mmio_region_write32(hmac->base_addr, HMAC_KEY0_REG_OFFSET + offset,
+                        key_part);
+  }
+}
+
```

Comment:
Not related to this PR, but I was wondering how you would feel about replacing this with `mmio_region_memcpy_to_mmio32` once #2066 lands.

Reply:
#2066 will make this much cleaner, I'll take note to refactor this once it gets merged, thanks!